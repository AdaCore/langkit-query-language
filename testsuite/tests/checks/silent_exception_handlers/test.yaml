driver: 'checker'
rule_name: silent_exception_handlers
input_sources: ['exc.adb']
arguments:
   silent_exception_handlers.subprograms: '["ada.exceptions.exception_message"]'
   silent_exception_handlers.subprogram_regexps: '["\.Log$"]'
