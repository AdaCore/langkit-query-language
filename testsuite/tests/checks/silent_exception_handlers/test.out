exc.adb:11:7: rule violation: silent exception handler
11 |        when others =>   --  FLAG
   | _______^
12 ||          null;
   ||______________^

exc_2.adb:11:4: rule violation: silent exception handler
11 |     when Constraint_Error =>  --  FLAG
   | ____^
   ||
   || ~~~ 8 other lines ~~~
   ||
20 ||       end if;
   ||_____________^

exc_2.adb:21:4: rule violation: silent exception handler
21 |     when Program_Error =>     --  FLAG
   | ____^
   ||
   || ~~~ 3 other lines ~~~
   ||
25 ||       end loop;
   ||_______________^

exc_2.adb:26:9: rule violation: silent exception handler
26 |    when E : others =>        --  FLAG
   |         ^

